var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py

```diff
@@ -196,41 +203,28 @@ def extend_config_flags(self):
         # TODO: it is unclear whether we should really use all elements of
         #  _ld_library_path because it looks like the only rpath that needs to be
         #  injected is self.component_prefix.linux.compiler.lib.intel64_lin.
-        flags_list = ["-Wl,-rpath,{}".format(d) for d in self._ld_library_path()]
+        common_flags = ["-Wl,-rpath,{}".format(d) for d in self._ld_library_path()]
+
+        # Make sure that underlying clang gets the right GCC toolchain by default
+        llvm_flags = ["--gcc-toolchain={}".format(self.compiler.prefix)]
+        classic_flags = ["-gcc-name={}".format(self.compiler.cc)]
```

Comment:
There's this annoying thing that:
1. `self.compiler.cc` can simply be say `gcc`, it's not guaranteed to be an absolute path -- and a module loaded with the compiler ensures PATH is set correctly.
2. Some environments such as Cray PE have a single compiler wrapper binary you're supposed to use for all compilers on the system, so that's what self.compiler.cc points to; environment variables make it dispatch to a particular compiler.

But I guess we can ignore that...

Reply:
Yup, so on a module-based system, users need to specify `modules: [gcc/x]` in Spack's `compilers.yaml` for the relevant intel compiler. (which can be confusing, but well...)

Regarding LLVM: we do what classic compilers do, but only once and for all during the installation.